app/apps/server/bridges/users.js

```diff
@@ -17,6 +22,33 @@ export class AppUserBridge {
 		return this.orch.getConverters().get('users').convertByUsername(username);
 	}
 
+	async create(user, appId) {
+		this.orch.debugLog(`The App ${ appId } is requesting to create a new user`);
+
+		user.password = 'pass';
+		user.requirePasswordChange = true;
+
+
+		if (typeof user.joinDefaultChannels === 'undefined') {
+			user.joinDefaultChannels = true;
+		}
+
+		Roles.findUsersInRole('admin').forEach((adminUser) => {
```

Comment:
Why the loop through each user who has an admin role? What if there are ten of them, wouldn't this code run multiple times and result in several users being defined or errors happening?

Also, @RocketChat/backend how should we handle the Apps creating users? Ideally we don't want to "associate" the user being created to an admin user who didn't actually create the new user.

Reply:
@graywolf336 while creating **rocket.cat** also uses same type of loop, but since the rocket.cat is created at begining there is only single admin,but in this case as you said there can be multiple admins, which may lead to errors. @graywolf336 can you please suggest how ideally the bots should be created in the server without getting errors.

Reply:
Until we hear back from @RocketChat/backend about what to do, just get the first admin user and use them.

Reply:
> Also, @RocketChat/backend how should we handle the Apps creating users? Ideally we don't want to "associate" the user being created to an admin user who didn't actually create the new user.

actually this admin user is just used for validation purposes.. for now it will not be associated and I think it is fine to do what @graywolf336 said, just get any admin and use it.